packages/table/src/components/Filter/Filter.js

```diff
@@ -0,0 +1,166 @@
+            id={filter.id}
+            name={filter.id}
+            options={filter.options}
+            value={filter.value}
+            onChange={(e) => handleFilter(filter.id, e.target.value)}
+            size={isGeneric ? "medium" : "small"}
+          />
+        );
+      case "datepicker":
+        return (
+          <Datepicker
+            key={filter.id}
+            label={filter.label}
+            id={filter.id}
+            name={filter.id}
+            value={filter.value}
+            onChange={(e) => handleFilter(filter.id, e.target.value)}
+            size={isGeneric ? "medium" : "small"}
+          />
+        );
+      case "telephone-number":
+        return (
+          <TelephoneNumber
+            key={filter.id}
+            label={filter.label}
+            id={filter.id}
+            name={filter.id}
+            value={filter.value}
+            onChange={(value) => handleFilter(filter.id, value)}
+            size={isGeneric ? "medium" : "small"}
+          />
+        );
+      default:
+        return (
+          <TextField
+            key={filter.id}
+            label={filter.label}
+            id={filter.id}
+            name={filter.id}
+            placeholder={filter.placeholder}
+            value={filter.value}
+            onChange={(e) => handleFilter(filter.id, e.target.value)}
+            size={isGeneric ? "medium" : "small"}
+          />
+        );
+    }
+  }
+
+  const handleColumnOrderChange = (columnIndex, direction) => {
+    const tempColumns = columns;
+    const selectedColumn = columns[columnIndex];
+    tempColumns.splice(columnIndex, 1);
+    tempColumns.splice(direction === 'up' ? columnIndex - 1 : columnIndex + 1, 0, selectedColumn);
+
+    columnOrderChanged(tempColumns);
+  }
+
+  const handleColumnVisibilityChange = (columnIndex) => {
+    const tempColumns = columns;
+    tempColumns[columnIndex].visible = !tempColumns[columnIndex].visible;
+    columnVisibilityChanged(tempColumns);
+  }
+
+  const handleFilter = (fieldName, value) => {
+    filtersChanged({
+      ...filters.reduce((acc, filter) => !!filter.value ? ({
+        ...acc,
+        [filter.id]: filter.value
+      }) : acc, {}),
+      [fieldName]: value
+    })
+  }
+
+  if (!genericFilter && !restFilters.length && disableColumnSorting) {
+    return <></>;
+  }
+
+  return (
+    <div className="m-table-filter">
+      <div className="m-table-filter__top-bar">
+        {genericFilter && (
+          <div className="m-table-filter__generic-filter">
+            {renderFilter(genericFilter, true)}
+          </div>
+        )}
+        {!!restFilters.length && (
+          <Button
+            iconRight={extraFiltersOpen ? "ai-arrow-up-1" : "ai-arrow-down-1"}
+            onClick={() => setExtraFiltersOpen(!extraFiltersOpen)}
+            className="u-margin-left-xs"
+            outline
+          >
+            {extraFilterMessage}
+          </Button>
+        )}
+        {!disableColumnSorting && (
```

Comment:
misschien dat hier nog extra controls toegevoegd zouden moeten kunnen worden door de gebruiker van de table component? in de smart table staat hier bvb ook nog een exportknop